### PR TITLE
tooling: Optimize `config_validation`

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/protodoc/manifest_to_json.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import json
+import pathlib
+import sys
+
+from google.protobuf import json_format
+
+from envoy.base.utils import ProtobufValidator
+
+sys.path = [p for p in sys.path if not p.endswith('bazel_tools')]
```
Comment (phlax) [inline tools/protodoc/manifest_to_json.py:9]: remove

**Comment by phlax** [comment]
This one gives a huge speed up to both the docs build and the config validation test

The bit im not certain about is that i have shifted the code for the validator to envoy.base.utils and im not sure how that will work with the tools image - ill check further on this, and potentially separate this change

**Comment by phlax** [comment]
cc @mattklein123 ^^

**Comment by phlax** [comment]
from checking the code, im reasonably confident this doesnt break anything

/wait for #21576

**Comment by phlax** [comment]
it would be great to land this one, it speeds CI up significantly

**Review by keith — APPROVED** [review]

**Pull request merged**